addons/sale_project/models/sale_order_line.py

```diff
@@ -76,12 +76,17 @@ def default_get(self, fields):
 
     @api.model
     def name_create(self, name):
+        ensure_is_service_product = False
         # To get the right product when creating a SOL on the fly, we need to get
         # the name that was entered in the field from the `default_get` method.
         # The easiest way of doing that is to store it in the context.
         if self.env.context.get('form_view_ref') == 'sale_project.sale_order_line_view_form_editable' and not self.env.context.get('action_view_sols'):
             self = self.with_context(sol_product_name=name)
-        return super().name_create(name)
+            ensure_is_service_product = True
+        sale_order_line = super().name_create(name)
+        if ensure_is_service_product and not self.browse(sale_order_line[0]).is_service:
```

Comment:
since the `name_create` does not return a recordset, we could call the variable `res` or `result` and check if the result is not falsy before browsing the record.

```suggestion
        result = super().name_create(name)
        if ensure_is_service_product and result and not self.browse(result[0]).is_service:
```